feature-libs/quote/components/quote-actions-by-role/quote-actions-by-role.component.spec.ts

```diff
@@ -192,20 +224,75 @@ describe('QuoteActionsByRoleComponent', () => {
         { type: QuoteActionType.CANCEL, isPrimary: false },
       ],
     };
+    const confirmationContextForSubmitAction: ConfirmationContext = {
+      quote: newMockQuoteWithSubmitAction,
+      title: 'quote.confirmActionDialog.buyer.submit.title',
+      confirmNote: 'quote.confirmActionDialog.buyer.submit.confirmNote',
+      successMessage: 'quote.confirmActionDialog.buyer.submit.successMessage',
+    };
     mockQuoteDetails$.next(newMockQuoteWithSubmitAction);
     fixture.detectChanges();
+    component.onClick(QuoteActionType.SUBMIT, newMockQuoteWithSubmitAction);
+    expect(launchDialogService.openDialog).toHaveBeenCalledWith(
+      LAUNCH_CALLER.ACTION_CONFIRMATION,
+      component.element,
+      component['viewContainerRef'],
+      { confirmationContext: confirmationContextForSubmitAction }
+    );
+  });
+
+  it('should open confirmation dialog when action is EDIT and state is BUYER_OFFER', () => {
+    spyOn(launchDialogService, 'openDialog');
+    const newMockQuoteWithEditActionAndBuyerOfferState: Quote = {
+      ...mockQuote,
+      allowedActions: [
+        { type: QuoteActionType.SUBMIT, isPrimary: true },
+        { type: QuoteActionType.CANCEL, isPrimary: false },
+        { type: QuoteActionType.EDIT, isPrimary: false },
+      ],
+      state: QuoteState.BUYER_OFFER,
+    };
+    const confirmationContextForSubmitAction: ConfirmationContext = {
+      quote: newMockQuoteWithEditActionAndBuyerOfferState,
+      title: 'quote.confirmActionDialog.buyer_offer.edit.title',
+      confirmNote: 'quote.confirmActionDialog.buyer_offer.edit.confirmNote',
+      warningNote: 'quote.confirmActionDialog.buyer_offer.edit.warningNote',
+      validity: 'quote.confirmActionDialog.validity',
+    };
+    mockQuoteDetails$.next(newMockQuoteWithEditActionAndBuyerOfferState);
+    fixture.detectChanges();
     component.onClick(
-      QuoteActionType.SUBMIT,
-      newMockQuoteWithSubmitAction.code
+      QuoteActionType.EDIT,
+      newMockQuoteWithEditActionAndBuyerOfferState
     );
     expect(launchDialogService.openDialog).toHaveBeenCalledWith(
-      LAUNCH_CALLER.REQUEST_CONFIRMATION,
+      LAUNCH_CALLER.ACTION_CONFIRMATION,
       component.element,
       component['viewContainerRef'],
-      { quoteCode: newMockQuoteWithSubmitAction.code }
+      { confirmationContext: confirmationContextForSubmitAction }
     );
   });
 
+  it('should not open confirmation dialog when action is EDIT and state is BUYER_DRAFT', () => {
+    spyOn(launchDialogService, 'openDialog');
+    const newMockQuoteWithEditActionAndBuyerDraftState: Quote = {
```

Comment:
```suggestion
    const quoteInBuyerDraftState: Quote = {
```
this object is not a 'Mock'

Reply:
done